app/wiring/AppComponents.scala

```diff
@@ -21,13 +24,33 @@ class AppComponents(context: Context, stage: String) extends BuiltInComponentsFr
     GoogleAuthConfig(clientId, clientSecret, redirectUrl, domain, antiForgeryChecker = AntiForgeryChecker.borrowSettingsFromPlay(httpConfiguration))
   }
 
+  private val googleGroupChecker = {
+    val s3URI = new AmazonS3URI(configuration.get[String]("googleAuth.serviceAccount.certificateS3Path"))
+    val stream = S3.s3Client
+      .getObject(s3URI.getBucket, s3URI.getKey)
+      .getObjectContent
+
+    val googleServiceAccountCredential = GoogleCredential.fromStream(stream)
+    stream.close()
+
+    val serviceAccount = GoogleServiceAccount(
+      googleServiceAccountCredential.getServiceAccountId,
+      googleServiceAccountCredential.getServiceAccountPrivateKey,
+      configuration.get[String]("googleAuth.serviceAccount.impersonatedUser"),
```

Comment:
what's an impersonatedUser ?

Reply:
https://github.com/guardian/play-googleauth#implement-googlegroups-based-access-control-using-the-directory-api

Reply:
added a link in a comment